ext/dom/token_list.c

```diff
@@ -83,6 +86,53 @@ static void dom_ordered_set_parser(HashTable *token_set, const char *position)
 	 *    => That's the token set. */
 }
 
+/* This returns true if all tokens in "token_set" are found in "value". */
+bool dom_ordered_set_all_contained(HashTable *token_set, const char *value, bool to_lowercase)
+{
+	/* This code is conceptually close to dom_ordered_set_parser(),
+	 * but without building a hash table.
+	 * Since the storage of the token set maps a value on itself,
+	 * we can reuse that storage as a "seen" flag by setting it to NULL. */
+	zval *zv;
+
+	uint32_t still_needed = zend_hash_num_elements(token_set);
+
+	value += strspn(value, ascii_whitespace);
+
+	while (*value != '\0' && still_needed > 0) {
+		const char *start = value;
+		value += strcspn(value, ascii_whitespace);
+		size_t length = value - start;
+
+		if (to_lowercase) {
+			ALLOCA_FLAG(use_heap)
+			char *lc_str = zend_str_tolower_copy(do_alloca(length + 1, use_heap), start, length);
+			zv = zend_hash_str_find(token_set, lc_str, length);
+			free_alloca(lc_str, use_heap);
```

Comment:
I thought we had a hash API for that, but we only have `zend_hash_str_find_ptr_lc` :/